zarr/hierarchy.py

```diff
@@ -638,17 +638,19 @@ def _array_iter(self, keys_only, method, recurse):
         else:
             dir_name = meta_root + self._path
             array_sfx = '.array' + self._metadata_key_suffix
+            group_sfx = '.group' + self._metadata_key_suffix
+
             for key in sorted(listdir(self._store, dir_name)):
                 if key.endswith(array_sfx):
                     key = key[:-len(array_sfx)]
+                    _key = key.rstrip("/")
+                    yield _key if keys_only else (_key, self[key])
+
                 path = self._key_prefix + key
                 assert not path.startswith("meta/")
-                if key.endswith('.group' + self._metadata_key_suffix):
+                if key.endswith(group_sfx):
                     # skip group metadata keys
                     continue
-                if contains_array(self._store, path):
-                    _key = key.rstrip("/")
-                    yield _key if keys_only else (_key, self[key])
```

Comment:
Joe and I were looking at this code together, and we couldn't understand the circumstances under which it would be necessary to call `contains_array` here. AFAICT, the V3 spec makes this unnecessary; if you've got an array metadata document, then there is an array at that path. It feels like something leftover from the V2 implementation.

Are we missing something?

Reply:
Reading the change & explanation again, it makes sense to me. I imagine it's safe to get in with the rest of the work that will be going on on the V3 side.

Reply:
@grlee77 - do you have thoughts on this change?